src/Illuminate/Validation/Concerns/ValidatesAttributes.php

```diff
@@ -2247,4 +2255,24 @@ protected function shouldBeNumeric($attribute, $rule)
             $this->numericRules[] = $rule;
         }
     }
+
+    /**
+     * Check the given value and loop over each file then return the result.
+     *
+     * @param  string  $functionName
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @param  array<int, int|string>  $parameters
+     * @return bool
+     */
+    protected function checkMultiValues($functionName, $attribute, $value, $parameters)
+    {
+        $isOk = true;
+
+        foreach ($value as $file) {
+            $isOk = $isOk && $this->$functionName($attribute, $file, $parameters);
+        }
+
+        return $isOk;
+    }
```

Comment:
You can skip the temporary variable:

~~~php
protected function checkMultiValues($functionName, $attribute, $value, $parameters)
{
    foreach ($value as $file) {
        if (! $this->{$functionName}($attribute, $file, $parameters)) {
            return false;
        }
    }

    return true;
}
~~~

Reply:
Nope, because when `$this->{$functionName}($attribute, $file, $parameters)` returns **false** this will exit from the loop then will return **true** and this is invalid approach so, I created a temporary variable to store all check value then return the final result.

Reply:
No it won't. A `return` will always exit the function it belongs to. Maybe you are confusing with a `break` statement.

This approach is used in many places on the framework:

https://github.com/laravel/framework/blob/278c6d7fd04be005c1175a7eaea868976177600c/src/Illuminate/Auth/SessionGuard.php#L457-L466

https://github.com/laravel/framework/blob/278c6d7fd04be005c1175a7eaea868976177600c/src/Illuminate/Auth/Access/Gate.php#L109-L120

https://github.com/laravel/framework/blob/278c6d7fd04be005c1175a7eaea868976177600c/src/Illuminate/Collections/Collection.php#L541-L552

https://github.com/laravel/framework/blob/278c6d7fd04be005c1175a7eaea868976177600c/src/Illuminate/Collections/Collection.php#L560-L575

these were the first results from searching Laravel code base for `return false`

Reply:
Returning in first failure will also skip checking the remaining items of the array. So if the first check fails and you have 30 items in your array, you won't need to check all of them.

Reply:
You are right maybe I am confused BETWEEN `return` and `break`.